src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,142 @@
+package duke;
+
+import duke.task.Task;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+
+import java.io.*;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+class Storage {
+
+    private String filePath;
+
+    /**
+     * Constructs a Storage object with a custom file path.
+     *
+     * @param filePath file path of the save file
+     */
+    Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Constructs a Storage object with a default file path.
+     *
+     */
+    Storage() {
+        this.filePath = getSaveFilePath();
+    }
+
+    /**
+     * Get the default file path specific to platform.
+     * @return default file path specific to platform
+     */
+    private String getSaveFilePath() {
+        URL url = getClass().getResource("/storage/data.json");
+        return url.getPath();
+        /*
+        if (System.getProperty("os.name").equals("Windows 10")) {
+            return "/Users/uicfa/Downloads/data.json";
+        } else {
+            return "/Users/leo/Downloads/data.json";
+        }
+         */
+    }
+
+    /**
+     * Creates the save file (used when it doesn't exist).
+     * @throws IOException
+     */
+    private void createSaveFile() throws IOException {
+        File file = new File(this.filePath);
+        if (!file.createNewFile()) {
+            throw new IOException();
+        }
+        writeToSaveFile("{\"data\":[]}");
+    }
+
+    /**
+     * Read and parse the save file.
```

Comment:
There should be an empty line between the description and parameters

Reply:
Great observation! I agree.